### PR TITLE
Pass the location of the data directory as an argument

### PR DESCRIPTION
**Pull request by papanikos**
Hi,

* Motivation
Point to a custom location where the downloaded `data` directory is stored.

* Main changes:
  - I moved the whole argument parsing into a new module `src/args.py` so it can be imported from `src/Variables.py` where the `_path` is defined. I retained the default behavior where `data` is expected to be found in the same directory as the main script, so it shouldn't mess with existing invocations in scripts/pipelines.
  - Added the `-d` argument to point to a data directory. Should be the one after downloading and extracting.
  - The main script is now executable and can be appended to the `$PATH` env variable (at least in Linux systems)

* Minor changes
The rest is mainly code formatting and some minor fixes here and there to play well with the new `args` module.
  - Be explicit about fasta suffixes (`.fasta`, `.fa`, `.fna`) when looping in the query dir (affects `src/s2star.py` and `src/crispr.py`)
  - Output dir is created if it doesn't exist
  - Include a warning about database size when downloading the pre-built files in README

* Miscellaneous
Tested this in a conda env with `python==3.9.1` and all dependencies installed through `conda`.
```
$ conda create -c bioconda -n vhmnet biopython numpy pandas blast
$ conda activate vhmnet
(vhmnet)$ CC=g++ python install setup.py --install-platlib=./src/
(vhmnet)$ export PATH=/path/to/VirHostMatcher-Net:$PATH
(vhmnet)$ VirHostMatcher-Net.py -q test/VGs -o output_VGs -t 8 -d /home/papanikos2/databases/vhm_net/data -i tmp
Specified output directory does not exist!
Creating /home/papanikos/Tools/vhmn/output_VGs
Loading packages...
Intermediate results will be stored in  /home/papanikos/Tools/vhmn/tmp
----Calculation of s2* is split into three parts----
----Start calculating s2* part I... ----
----Finished calculating s2* part I----
----Start calculating s2* part II... ----
----Finished calculating s2* part II----
----Start calculating s2* part III... ----
----Finished calculating s2* part III----
----Start calculating network neighborhood feature values...----
----Finished Calculating network neighborhood feature values----
----Calculating crispr feature values for  DQ517338.fasta  ----
----Calculating crispr feature values for  KF669661.fasta  ----
----Calculating crispr feature values for  KC821607.fasta  ----
----Calculating crispr feature values for  JX878496.fasta  ----
----CRISPR intermediate files are stored in  /home/papanikos/Tools/vhmn/tmp/CRISPR/  ----
Writing feature scores to /home/papanikos/Tools/vhmn/output_VGs/feature_values...
Calculating prediction scores...
Making predictions...
---- Predictions are written to  /home/papanikos/Tools/vhmn/output_VGs  ----
```

Glad to take up on any comments!

### CODE REVIEW

**Comment by papanikos** [comment]
Oh, and the reformatted help

```
(vhmnet)$ VirHostMatcher-Net.py -h
usage: VirHostMatcher-Net.py [-h] -q QUERY_VIRUS_DIR -o OUTPUT_DIR [-t NUM_THREADS] [--short-contig] [-n topN] [-i INTERMEDIATE_DIR] [-l GENOME_LIST] [-d DATA_DIR]

VirHostMatcher-Net: A network-based tool for predicting hosts given query viruses

Required Arguments:
  -q QUERY_VIRUS_DIR   Directory containing query virus genomes with .fasta, .fa or .fna suffixes. One genome per file. (default: None)
  -o OUTPUT_DIR        Output directory. It is created if it doesn't exist (default: None)

Optional Arguments:
  -h, --help           show this help message and exit
  -t NUM_THREADS       Number of threads to use. (default: 1)
  --short-contig       Predict hosts for short viral contigs. (WIsH model files are required in this mode (default: False)
  -n topN              Number of top predictions written to the output files. All predictions will be output if there is a tie in score (default: 1)
  -i INTERMEDIATE_DIR  Directory storing intermediate result. (default: ./intermediate_res)
  -l GENOME_LIST       Location of the file containing host NCBI genome names of interest (default: None)
  -d DATA_DIR          Directory where models, blast and CRISPRs are located. (default: /home/papanikos/Tools/vhmn/data)
```

**Comment by papanikos** [comment]
@WeiliWw Just a straight-up mention to check whether you get notified. It is not _that_ pressing :smile:

File: src/crispr.py [hunk 1]
```diff
@@ -70,7 +70,8 @@ def crisprSingle(item, query_virus_dir, output_dir, numThreads):
 '''
 def crispr_calculator(query_virus_dir, output_dir, numThreads):
     query_cont = []
-    query_list = os.listdir(query_virus_dir)
+    query_list = [f for f in os.listdir(query_virus_dir) 
+                  if f.endswith(('.fasta', '.fa', '.fna'))]
```
Comment (WeiliWw) [inline src/crispr.py]: Consider using f.lower().endswith(...) instead?

File: src/s2star.py [hunk 2]
```diff
@@ -49,7 +49,8 @@ def get_f(a_K_count, a_expect):
 
 def get_all_f(Dir, K, order, Reverse, numThreads):
     M = order + 1
-    sequence_list = os.listdir(Dir)
+    sequence_list = [f for f in os.listdir(Dir) 
+                     if f.endswith(('.fasta', '.fa', '.fna'))]
```
Comment (WeiliWw) [inline src/s2star.py]: ditto

**Review by WeiliWw — CHANGES_REQUESTED** [review]
Hi Nikos, Thanks for contributing to this tool! The changes look great to me. Just one minor formatting thing and then it's ready to merge.

File: src/crispr.py (hunk 1 above)
Comment (papanikos) [reply src/crispr.py]: Done

**Review by WeiliWw — APPROVED** [review]
lgtm

**Pull request merged**